libbeat/cmd/instance/metrics.go

```diff
@@ -1,16 +1,39 @@
 package instance
 
 import (
+	"os"
 	"runtime"
 	"time"
 
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/monitoring"
+	sigar "github.com/elastic/gosigar"
+)
+
+type cpuSample struct {
+	time      time.Time
+	procTimes sigar.ProcTime
+}
+
+var (
+	numCores   = runtime.NumCPU()
+	lastSample = cpuSample{
+		time:      time.Now(),
+		procTimes: sigar.ProcTime{},
+	}
 )
 
 func init() {
+	pid := os.Getpid()
+	err := lastSample.procTimes.Get(pid)
+	if err != nil {
+		panic(err)
+	}
+
 	metrics := monitoring.Default.NewRegistry("beat")
 
 	monitoring.NewFunc(metrics, "memstats", reportMemStats, monitoring.Report)
+	monitoring.NewFunc(metrics, "cpu", reportCPU, monitoring.Report)
```

Comment:
If the Beat is running on a platform where we have no support for getting process CPU data maybe we should not register this metric?

@tsullivan I assume in the UI it would be best to tell the user that we don't have CPU usage data rather than showing 0. Would not sending the metric be the way to go?